CMakeLists.txt

```diff
@@ -240,12 +259,32 @@ if (ENABLE_MICROPROFILE)
             MICROPROFILE_ENABLED
             MICROPROFILE_GPU_TIMERS=0
     )
+    if(BUILD_TEST)
+        target_include_directories(
+                ${PROJECT_NAME}_Tests
+                PRIVATE
+                ${microprofile_SOURCE_DIR}
+        )
+        target_compile_definitions(
+                ${PROJECT_NAME}_Tests
+                PRIVATE
+                MICROPROFILE_ENABLED
+                MICROPROFILE_GPU_TIMERS=0
+        )
+    endif(BUILD_TEST)
 	if (WIN32)
 		target_link_libraries(
 				${PROJECT_NAME}
 				PRIVATE
 				ws2_32.lib
 		)
+    if(BUILD_TEST)
+        target_link_libraries(
+            ${PROJECT_NAME}_Tests
+            PRIVATE
+            ws2_32.lib
```

Comment:
Create a variable "ADDITIONAL_LIBS" and fill it condionally. that's easier to read
Like
```
set (ADDITIONAL_LIBS)
if (BUILD_TEST)
  list(APPEND ADDITIONAL_LIBS ${ADDITIONAL_LIBS} ws2_32.lib
endif()

        target_link_libraries(
                ${PROJECT_NAME}_Tests
                PRIVATE
                # need to link with mingw32 before SDL2
                mingw32
                ${ADDITIONAL_LIBRARIES}
        )

Reply:
target_link_libraries must be inside BUILD_TEST, because if !BUILD_TEST then we cannot link anything

Reply:
sry, i just took a brief look and overlooked that that's a different target